python_modules/libraries/dagster-aws/dagster_aws/ssm/resources.py

```diff
@@ -71,75 +105,78 @@ def example_job():
-                "all parameters with the given tag key will be returned. Note: the "
-                "underlying AWS API will throw an exception when more than 10 "
-                "parameters match the tag filter query."
-            ),
-        ),
-    }
-)
+class ParameterStoreTag(Config):
+    key: str = Field(description="Tag key to search for.")
+    values: Optional[List[str]] = Field(deafult=None, description="List")
 
 
-@resource(
-    merge_dicts(
-        BOTO3_SESSION_CONFIG,
-        {
-            "parameters": Field(
-                Array(str),
-                is_required=False,
-                default_value=[],
-                description="An array of AWS SSM Parameter Store parameter names to fetch.",
-            ),
-            "parameter_tags": Field(
-                Array(tag_shape),
-                is_required=False,
-                default_value=[],
-                description=(
-                    "AWS SSM Parameter store parameters with this tag will be fetched and made"
-                    " available."
-                ),
-            ),
-            "parameter_paths": Field(
-                [str],
-                is_required=False,
-                default_value=[],
-                description="List of path prefixes to pull parameters from.",
-            ),
-            "with_decryption": Field(
-                bool,
-                is_required=False,
-                default_value=False,
-                description=(
-                    "Whether to decrypt parameters upon retrieval. Is ignored by AWS if parameter"
-                    " type is String or StringList"
-                ),
-            ),
-            "add_to_environment": Field(
-                bool,
-                is_required=False,
-                default_value=False,
-                description="Whether to mount the parameters as environment variables.",
-            ),
-        },
+class ParameterStoreResource(Boto3ResourceBase[Dict[str, str]]):
+    parameters: List[str] = Field(
+        default=[], description="An array of AWS SSM Parameter Store parameter names to fetch."
     )
-)
+    parameter_tags: List[ParameterStoreTag] = Field(
+        default=[],
+        description=(
+            "AWS SSM Parameter store parameters with this tag will be fetched and made available."
+        ),
+    )
+    parameter_paths: List[str] = Field(
+        default=[], description="List of path prefixes to pull parameters from."
+    )
+    with_decryption: bool = Field(
+        default=False,
+        description=(
+            "Whether to decrypt parameters upon retrieval. Is ignored by AWS if parameter type is"
+            " String or StringList"
+        ),
+    )
+    add_to_environment: bool = Field(
+        default=False, description="Whether to add the parameters to the environment."
+    )
+
+    def create_resource(self, context: InitResourceContext) -> Any:
```

Comment:
for our own libraries we should consider typing the generators correctly per https://stackoverflow.com/a/70277752/21589729